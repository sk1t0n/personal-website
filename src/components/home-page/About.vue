<template>
  <section id="about">
    <h2>{{ $t('about.word1') }} <span>{{ $t('about.word2') }}</span></h2>
    <div class="row">
      <q-img :src="photoUrl" />
      <div>
        <p>{{ $t('about.p1') }}</p>
        <p>{{ $t('about.p2') }}</p>
        <p>{{ $t('about.p3') }}</p>
        <p>{{ $t('about.p4') }}</p>
        <p>{{ $t('about.p5') }}</p>
        <p>{{ $t('about.p6') }}</p>
      </div>
    </div>
  </section>
</template>

<script setup>
const photoUrl = 'images/photo.jpeg'
</script>

<style lang="scss">
#about {
  .row {
    margin-top: 20px;
    justify-content: center;

    .q-img {
      max-width: 256px;
      max-height: 256px;
    }

    div {
      padding-top: 0.6em;
    }

    p {
      font-size: 1em;
      margin-top: 10px;
      margin-bottom: 10px;
    }
  }
}

@media screen and (max-width: 767px) {
  #about {
    .row {
      flex-direction: column;
      justify-content: center;
      margin-left: 2em;
      margin-right: 2em;

      .q-img,
      div {
        margin: 0 auto;
      }

      p:first-child {
        margin-top: 20px;
      }
    }
  }
}

@media screen and (min-width: 768px) {
  #about {
    .row {
      .q-img {
        margin-right: 30px;
      }

      p:first-child {
        margin-top: 17px;
      }
    }
  }
}
</style>
