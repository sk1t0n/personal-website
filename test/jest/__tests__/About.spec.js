import { config, mount, createLocalVue } from '@vue/test-utils'
import About from '../../../src/components/home-page/About.vue'
import translations from '../../../src/i18n/en-us'
import * as All from 'quasar'
// import langEn from 'quasar/lang/en-us' // change to any language you wish! => this breaks wallaby :(
const { Quasar } = All

const components = Object.keys(All).reduce((object, key) => {
  const val = All[key]
  if (val && val.component && val.component.name != null) {
    object[key] = val
  }
  return object
}, {})

config.mocks['$t'] = (msg) => {
  const arr = msg.split('.')
  return translations[arr[0]][arr[1]]
}

describe('About.vue', () => {
  const localVue = createLocalVue()
  localVue.use(Quasar, { components }) // lang: langEn

  const wrapper = mount(About, {
    localVue
  })
  const vm = wrapper.vm

  it('header has correct text', () => {
    const text = 'About Me'
    expect(wrapper.find('h2').text()).toBe(text)
  })

  it('q-img has correct url', () => {
    const text = 'images/photo.jpeg'
    expect(vm.photoUrl).toBe(text)
  })

  it('correct snapshot', () => {
    expect(wrapper.text()).toMatchSnapshot()
  })
})
