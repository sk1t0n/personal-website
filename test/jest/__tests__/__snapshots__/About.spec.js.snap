// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About.vue correct snapshot 1`] = `"About Me  Fullstack Developer Programming languages: Python, JavaScript Backend: Django, Django REST framework, FastAPI Databases: PostgreSQL, MongoDB Frontend: Vue.js, Vuex, Quasar Framework, Buefy, Nuxt.js DevOps: Docker, Docker Compose, Ansible Other: Web scraping (Beautiful Soup, Scrapy, Selenium), Telegram bots (aiogram)"`;
